docs/how-to/h-deploy-eks.md

```diff
@@ -0,0 +1,232 @@
+* [eksctl](https://eksctl.io/installation/) - the Amazon EKS CLI
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) - the Amazon Web Services CLI
+
+To check they are all correctly installed, you can run the commands demonstrated below with sample outputs:
+
+```shell
+> juju version
+3.1.7-ubuntu-amd64
+
+> kubectl version --client
+Client Version: v1.28.2
+Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
+
+> eksctl info
+eksctl version: 0.159.0
+kubectl version: v1.28.2
+
+> aws --version
+aws-cli/2.13.25 Python/3.11.5 Linux/6.2.0-33-generic exe/x86_64.ubuntu.23 prompt/off
+```
+
+### Authenticate
+
+Create an IAM account (or use legacy access keys) and login to AWS:
+
+```shell
+> aws configure
+AWS Access Key ID [None]: SECRET_ACCESS_KEY_ID
+AWS Secret Access Key [None]: SECRET_ACCESS_KEY_VALUE
+Default region name [None]: eu-west-3
+Default output format [None]:
+
+> aws sts get-caller-identity
+{
+    "UserId": "1234567890",
+    "Account": "1234567890",
+    "Arn": "arn:aws:iam::1234567890:root"
+}
+```
+
+## Create a new EKS cluster
+
+Export the deployment name for further use:
+```shell
+export JUJU_NAME=eks-$USER-$RANDOM
+```
+
+This following examples in this guide will use the location `eu-west-3` and K8s `v.1.27` - feel free to change this for your own deployment.
+
+Sample `cluster.yaml`:
+
+```shell
+~$ cat <<-EOF > cluster.yaml
+---
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+    name: ${JUJU_NAME}
+    region: eu-west-3
+    version: "1.27"
+iam:
+  withOIDC: true
+
+addons:
+- name: aws-ebs-csi-driver
+  wellKnownPolicies:
+    ebsCSIController: true
+
+nodeGroups:
+    - name: ng-1
+      minSize: 3
+      maxSize: 5
+      iam:
+        attachPolicyARNs:
+        - arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy
+        - arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy
+        - arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - arn:aws:iam::aws:policy/AmazonS3FullAccess
+      instancesDistribution:
+        maxPrice: 0.15
+        instanceTypes: ["m5.xlarge", "m5.2xlarge"] # At least two instance types should be specified
+        onDemandBaseCapacity: 0
+        onDemandPercentageAboveBaseCapacity: 50
+        spotInstancePools: 2
+EOF
+```
+
+Bootstrap EKS cluster with the following command:
+
+```shell
+eksctl create cluster -f cluster.yaml
+```
+
+Sample output:
+
+```shell
+...
+2023-10-12 11:13:58 [ℹ]  using region eu-west-3
+2023-10-12 11:13:59 [ℹ]  using Kubernetes version 1.27
+...
+2023-10-12 11:40:00 [✔]  EKS cluster "eks-taurus-27506" in "eu-west-3" region is ready
+```
+
+## Bootstrap Juju on EKS
+
+Add Juju K8s clouds:
+
+```shell
+juju add-k8s $JUJU_NAME
+```
+
+Bootstrap Juju controller:
+
+```shell
+juju bootstrap $JUJU_NAME
+```
+
+## Deploy Charms
+
+Create a new Juju model, if needed:
+
+```shell
+juju add-model <MODEL_NAME>
```

Comment:
provide a working example. This command as typed will fail

Reply:
it is expected, as it is a how-to and not a tutorial.